### PR TITLE
Add cache observer

### LINKED ISSUE #756: Add Prometheus support for Cache Providers
The Cache metrics will be prefixed with `tegola_cache_` and the following things will be observed

Each one will have a label of `type` that will be the cache type name

* misses_total is a counter of the total number of misses
* hits_total is a counter of the total number of hits
* flight_request is a gauge of the cache requests currently being served
* duration_seconds is a histogram of the duration of the current get cache requests, partitioned by map_name/map_layer, and z. 
* response_size_bytes is a histogram of response sized in bytes with the following buckets, 500kb, 1 KB, 10 KB, and 1 MB.
* errors is the number of errors from a get operation (that is not a miss error)
* set_duration_seconds is a histogram of the duration for filling the cache after a miss.
* set_errors is the number of errors from a get operation

### PR DESCRIPTION
**Pull request by gdey**
Add Observability metrics for Cache Providers.

This will work for all providers.
Includes support for push. 

fixes #756

### CODE REVIEW

**Comment by gdey** [comment]
I have not had a chance to test the push on this. So, this is not quite ready.

File: observability/observability.go [hunk 1]
```diff
@@ -5,9 +5,19 @@ import (
...
+
 	"github.com/go-spatial/tegola/dict"
 )
 
+const (
+	ObserveVarMapName   = ":map_name"
+	ObserveVarLayerName = ":layer_name"
+	ObserveVarX         = ":x"
+	ObserveVarY         = ":y"
+	ObserveVarZ         = ":z"
```
Comment (ARolek) [inline observability/observability.go:18]: Maybe name this ObserveVarTileX, ObserveVarTileY, ObserveVarTileZ

File: observability/prometheus/cache.go [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+			lbs["y"] = strconv.FormatInt(int64(key.Y), 10)
+		case observability.ObserveVarZ:
+			lbs["z"] = strconv.FormatInt(int64(key.Z), 10)
+		}
+	}
+	lbs["sub_command"] = cmd
+	return lbs
+}
+
+// Get
```
Comment (ARolek) [inline observability/prometheus/cache.go:121]: Missing rest of comment.

File: observability/prometheus/cache.go [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+	} else {
+		co.missesCounter.With(lbs).Add(1)
+	}
+
+	co.responseSizeBytes.With(lbs).Observe(float64(len(body)))
+	co.inFlightGauge.Dec()
+	return body, ok, nil
+}
+
+// Set
```
Comment (ARolek) [inline observability/prometheus/cache.go:144]: incomplete comment.

File: observability/prometheus/cache.go [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+		co.errors.With(lbs).Add(1)
+		co.inFlightGauge.Dec()
+		return err
+	}
+	co.responseSizeBytes.With(lbs).Observe(float64(len(body)))
+	co.inFlightGauge.Dec()
+	return nil
+}
+
+// Purge will record the
```
Comment (ARolek) [inline observability/prometheus/cache.go:161]: incomplete comment.

**Review by ARolek — COMMENTED** [review]
A few comments inline. This is looking great! 

Per the discussion today we need a way for push to periodically send metric updates.

**Pull request merged**